### PR TITLE
Link with subresource

### PR DESCRIPTION
**Pull request by jdemangeon**

### CODE REVIEW

File: src/app/custom/translations.tsv [hunk 1]
```diff
@@ -204,6 +204,7 @@
 "lodexField"	"Lodex - Field"	"Lodex - Champ"
 "hidden_info"	"Hide the resource field"	"Cacher le champ de la ressource"
 "a_column"	"A value from an existing column"	"Une valeur en provenance d'une colonne existante"
+"a_subresource_uri"	"A value from subresource uri"	"Une valeur en provenance d'une uri de subresource"
```
Comment (JulienMattiussi) [inline src/app/custom/translations.tsv]: I think it's "sous-ressource" in french ;)

File: src/app/custom/translations.tsv [hunk 2]
```diff
@@ -533,5 +534,6 @@
 "import_file_text"	"Drop your file on this page"	"Glisser un fichier sur cette page"
 "new_subresource"	"New subresource"	"Nouvelle sous-ressource"
 "delete"	"Delete"	"Supprimer"
-"confirm_delete_subresource"	"This subresource will be definitively deleted, Are you sure?"	"Cette sous-resource va être supprimée. Êtes-vous sûr de vouloir continuer ?"
+"confirm_delete_subresource"	"This subresource and all its fields will be definitively deleted, Are you sure?"	"Cette sous-resource ainsi que tous les champs associés vont êtres supprimées. Êtes-vous sûr de vouloir continuer ?"
```
Comment (JulienMattiussi) [inline src/app/custom/translations.tsv]: I think it's "sous-ressource" in french with two "s"
and "supprimées" should be "supprimés"

File: src/app/custom/translations.tsv [hunk 3]
```diff
@@ -533,5 +534,6 @@
...
 "no_field_for_subresource"	"No field set for this Subresource"	"Aucun champs renseigné pour cette sous-ressource"
+"transformer_no_editable_with_subresource_uid_value"	"Transformers can't be edited when subresource uid is selected"	"Les transformers ne peuvent êtres édités quand la valeur choisie est un uid de sous-ressource"
```
Comment (JulienMattiussi) [inline src/app/custom/translations.tsv:539]: this one is well written ;)

File: src/app/js/fields/wizard/index.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { connect } from 'react-redux';
 import compose from 'recompose/compose';
 import IconButton from '@material-ui/core/IconButton';
 import CloseIcon from '@material-ui/icons/Close';
-import { hideAddColumns } from '../../admin/parsing';
+import { formValueSelector } from 'redux-form';
```
Comment (JulienMattiussi) [inline src/app/js/fields/wizard/index.js]: Unused here

File: src/app/js/fields/wizard/index.js [hunk 5]
```diff
@@ -30,6 +32,7 @@ import StepSearch from './StepSearch';
 import StepSemantics from './StepSemantics';
 import FieldExcerpt from '../../admin/preview/field/FieldExcerpt';
 import Actions from './Actions';
+import { isSubresourceTransformation } from './StepValueSubresource';
```
Comment (JulienMattiussi) [inline src/app/js/fields/wizard/index.js]: Unused here

File: src/app/js/fields/wizard/StepValueSubresource.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+import translate from 'redux-polyglot/translate';
+import compose from 'recompose/compose';
+import withHandlers from 'recompose/withHandlers';
+import withState from 'recompose/withState';
+import { connect } from 'react-redux';
+import { formValueSelector } from 'redux-form';
+
+import { FIELD_FORM_NAME } from '../';
+import { polyglot as polyglotPropTypes } from '../../propTypes';
+import SelectSubresourceField from './SelectSubresourceField';
```
Comment (JulienMattiussi) [inline src/app/js/fields/wizard/StepValueSubresource.js]: Unused here

File: src/api/models/field.js [hunk 7]
```diff
@@ -140,6 +140,14 @@ export default async db => {
     collection.removeById = id =>
         collection.remove({ _id: new ObjectID(id), name: { $ne: 'uri' } });
 
+    collection.removeBySubresource = subresourceId =>
+        collection.remove({
+            $or: [
+                { subresourceId: new ObjectID(subresourceId) },
+                { subresourceId },
```
Comment (JulienMattiussi) [inline src/api/models/field.js:147]: I'm surprised that works without an error in mongo, depending on the type of the variable ou are giving.
Are to two cases really used ?

Comment (jdemangeon) [reply src/api/models/field.js:147]: Yes, if not it doesn't delete one of those

**Review by JulienMattiussi — APPROVED** [review]

**Pull request merged**